internal/handlers/sponsor_handler.go

```diff
@@ -0,0 +1,96 @@
+package handlers
+
+import (
+	"backend/internal/middleware"
+	"backend/internal/models"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
+)
+
+type SponsorHandler struct {
+	db *gorm.DB
+}
+
+func NewSponsorHandler(db *gorm.DB) *SponsorHandler {
+	return &SponsorHandler{db: db}
+}
+
+func (h *SponsorHandler) Register(r *gin.RouterGroup) {
+	sponsor := r.Group("/sponsor")
+	{
+		// Public endpoints (require auth)
+		sponsor.GET("", h.List)
+		sponsor.GET("/:id", h.Get)
+
+		// Admin-only endpoints
+		admin := sponsor.Group("/admin")
```

Comment:
same as above: no need to group with sepearate base URL for admin routes, just make sure the middleware `AdminRequired()` is there for the POST, PUT and DELETE endpoints